prov/verbs/src/verbs_info.c

```diff
@@ -260,8 +259,8 @@ int fi_ibv_check_rx_attr(const struct fi_rx_attr *attr,
 
 	compare_mode = attr->mode ? attr->mode : hints->mode;
 
-	check_mode = FI_IBV_EP_TYPE_IS_RDM(info) ? VERBS_RDM_MODE :
-		(hints->caps & FI_RMA) ? info->rx_attr->mode : VERBS_MODE;
+	check_mode = (hints->caps & FI_RMA) ? info->rx_attr->mode :
+		(info->rx_attr->mode & ~FI_RX_CQ_DATA);
```

Comment:
This line disables requiring FI_RX_CQ_DATA mode if the app doesn't use FI_RMA. VERBS_RX_MODE could potentially include any other mode that doesn't concern FI_RMA. Although we don't have such a thing now.